### PR TITLE
Fixup base fee type

### PR DESCRIPTION
**Pull request by 3commascapital**
## Description

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../CONTRIBUTIONS.md).

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run lint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:unit` with success.
- [x] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

**Comment by 3commascapital** [comment]
documentation does not match, a bigint should be returned

https://github.com/web3/web3.js/blob/4.x/docs/docs/guides/web3_upgrade_guide/1.x/web3_eth_migration_guide.md#web3ethgetblock

**Comment by 3commascapital** [comment]
was build process recently modified? not sure why no jobs were run.  @avkos , @jdevcs , @luu-alex, @Muhammad-Altabba  anything look odd to you?

**Comment by luu-alex** [comment]
@3commascapital , test suites need to be approved first, its running now

**Comment by 3commascapital** [comment]
seems like it's just node connection issues? not sure how to debug this

**Comment by Muhammad-Altabba** [comment]
Hi @3commascapital 
You just need to press on the `Details` of the failed job, like this:
Screenshot from 2023-09-27 08-09-31

---

And, then you can see the command that you can also run on your machine:

failing-job

---

 And if you scroll you can see the logs:
image

**Comment by Muhammad-Altabba** [comment]
Hello @jdevcs, 
Do you think we can update the types? Or do we need to update the documentation?
Thanks,

**Comment by jdevcs** [comment]
> Hello @jdevcs, Do you think we can update the types? Or do we need to update the documentation? Thanks,

Its fix in recently merged PR.

**Comment by 3commascapital** [comment]
unfrotunately when i run these locally, the task stalls out. this particular script has been running for 10 mins and has not moved forward: 
```
➜  web3.js git:(fixup-base-fee-type) yarn test:e2e:ganache:ws:firefox
yarn run v1.22.15
$ ./scripts/test-runner.sh ganache ws firefox
Node software used for tests:  ganache
Node running on:  ws://127.0.0.1:8545
$ WEB3_SYSTEM_TEST_BACKEND=ganache && ./scripts/ganache.sh start 1
Starting ganache ...
2b20df245e064a2fef0ab9a698dcd7fd3430dc0cd1a5e108ea1852b6f04867c2
docker: Error response from daemon: driver failed programming external connectivity on endpoint inspiring_sanderson (bb0b515e76a372b03dcf811d0eb3459687790375688db7527d793f894e104875): Bind for 0.0.0.0:8545 failed: port is already allocated.
Waiting for ganache...
Waiting for localhost:8545.
Connected!
Ganache started...
$ node ./scripts/gen_accounts.js
$ ~/web3/web3.js/node_modules/.bin/lerna run test:e2e:firefox --stream
lerna notice cli v6.6.2
lerna info versioning independent

 >  Lerna (powered by Nx)   Running target test:e2e:firefox for 3 projects:

    - web3-eth-contract
    - web3-eth
    - web3-providers-http

 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

> web3-providers-http:"test:e2e:firefox"

web3-providers-http: $ npx cypress run --headless --browser firefox
web3-providers-http: Need to install the following packages:
web3-providers-http:   cypress@13.2.0

```

**Comment by luu-alex** [comment]
@3commascapital  looks like you fixed ur textcases, we have some inconsistent test cases that are failing not related to you.

File: packages/web3-core/src/formatters.ts [hunk 1]
```diff
@@ -450,7 +450,7 @@ export const outputBlockFormatter = (block: BlockInput): BlockOutput => {
 	}
 
 	if (block.baseFeePerGas) {
-		modifiedBlock.baseFeePerGas = hexToNumber(block.baseFeePerGas);
+		modifiedBlock.baseFeePerGas = outputBigIntegerFormatter(block.baseFeePerGas);
```
Comment (jdevcs) [inline packages/web3-core/src/formatters.ts:453]: This function `outputBlockFormatter` is marked for deprecated.  We can merge this but will encourage not to use it, and use format function in web3 utils. example for block formatting: https://github.com/web3/web3.js/blob/b8fa7124d1d1e4c9c16b60ad2ce58559f72efd27/packages/web3-eth/src/rpc_method_wrappers.ts#L248

**Review by jdevcs — APPROVED** [review]

File: packages/web3-core/src/formatters.ts (hunk 1 above)
Comment (jdevcs) [reply packages/web3-core/src/formatters.ts:453]: I pointed changed line but was mentioning function containing it.

**Review by luu-alex — APPROVED** [review]

**Comment by 3commascapital** [comment]
just wanted to bump this if there is anyone else that needs to approve to move it forward

**Comment by luu-alex** [comment]
@3commascapital  Hey there, thanks for the patience. I think the last thing you need to do is update the [changelog](https://github.com/web3/web3.js/blob/0004f51703097b4e1a06348ff3213f5735fb4757/packages/web3-core/CHANGELOG.md) with your change and it should be good to go

**Comment by 3commascapital** [comment]
thanks @luu-alex . done!

File: packages/web3-core/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "web3-core",
-	"version": "4.2.0",
+	"version": "4.2.0-bigint",
```
Comment (luu-alex) [inline packages/web3-core/package.json]: we will change the version number, would be great if u can change it back

File: packages/web3-core/CHANGELOG.md [hunk 3]
```diff
@@ -183,4 +183,5 @@ Documentation:
 
 ### Changed
 
--   defaultTransactionType is now type 0x2 instead of 0x0 (#6282)
\ No newline at end of file
+-   defaultTransactionType is now type 0x2 instead of 0x0 (#6282)
+-   Allows formatter to parse large base fee
```
Comment (luu-alex) [inline packages/web3-core/CHANGELOG.md]:
```suggestion
-   Allows formatter to parse large base fee (#6456)
```

**Comment by luu-alex** [comment]
2 more changes needed, thanks alot for ur contributions

**Comment by 3commascapital** [comment]
❤️ thank you!

**Pull request merged**